src/pages/iou/MoneyRequestAmountForm.tsx

```diff
@@ -108,7 +105,7 @@ function MoneyRequestAmountForm(
     const isFocused = useIsFocused();
     const wasFocused = usePrevious(isFocused);
 
-    const formattedTaxAmount = CurrencyUtils.convertToDisplayString(Math.abs(taxAmount), currency);
+    const formattedTaxAmount = CurrencyUtils.convertToDisplayString(Math.abs(taxAmount), moneyRequestAmountInput?.current?.getCurrency());
```

Comment:
Hmm, looks we are running into a problem, what's the source of truth, the `currency` prop or the `getCurrency` method?

Reply:
`getCurrency` method, I see this problem when we diplsay the limit error for tax amount.

Reply:
How can we make this easier to understand? Most importantly when to use `currency` vs `getCurrency`

Reply:
The bug happens when we paste another currency and then enter the tax amount that is over limit then the error displays with the old currency if we use `currency`.

What do you think about renaming this to `currencyFromProp`?

Reply:
I think it's still unclear when ti use one vs the other. e.g. in `initializeAmount` shouldn't we use the currency from the input? (the function is called on tab switch)